module8/Homework8/Homework7/Views/APIListViewModel.swift

```diff
@@ -0,0 +1,65 @@
+/// Copyright (c) 2024 Kodeco Inc.
+/// 
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+/// 
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+/// 
+/// Notwithstanding the foregoing, you may not use, copy, modify, merge, publish,
+/// distribute, sublicense, create a derivative work, and/or sell copies of the
+/// Software in any work that is designed, intended, or marketed for pedagogical or
+/// instructional purposes related to programming, coding, application development,
+/// or information technology.  Permission for such use, copying, modification,
+/// merger, publication, distribution, sublicensing, creation of derivative works,
+/// or sale is expressly withheld.
+/// 
+/// This project and source code may use libraries or frameworks that are
+/// released under various Open-Source licenses. Use of those libraries and
+/// frameworks are governed by their own individual licenses.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import Foundation
+import Combine
+
+class APIListViewModel: ObservableObject {
+  @Published var apiStore: APIStore
+  @Published var showingAPIErrorView = false
+  @Published var showingAPILoadingIndicator = false
+  var apiDataStateCancellable: AnyCancellable?
+
+  init(
+    apiStore: APIStore,
+    showingAPIErrorView: Bool = false,
+    showingAPILoadingIndicator: Bool = false
+  ) {
+    self.apiStore = apiStore
+    self.showingAPIErrorView = showingAPIErrorView
+    self.showingAPILoadingIndicator = showingAPILoadingIndicator
+
+    apiDataStateCancellable = apiStore.$apiDataState.sink { [weak self] state in
```

Comment:
Getting a bit ahead of things w/ this Combine subscriptions (still not sure exactly how these work). Perhaps there's a simpler way to accomplish this w/ out Combine?

Reply:
Combine is ok here. It's a little complicated for what you want. I started trying to change it, but didn't want to mess everything up. I think you can greatly simplify by throwing out the APIListViewModel and just using APIStore directly and having the view react to the `apiDataState`.

Reply:
Its definitely worth digging in or at least talking through how to simplify. 

So the problem I ran into w/ having the view use apiStore.apiDataState directly, was that I wanted to have a binding "@Binding var showingErrorView: Bool" to pass into ErrorSheet to allow it to dismiss itself. So the question was how to connect state updates from apiStore.apiDataState such that updates to it would update showingErrorView, but that showingErrorView could also be updated independently (when ErrorSheet dismisses itself).

The view model basically serves as a place to do this mapping. Maybe there's a simpler way to do this inline in the view?